DEV-INTRO.md

```diff
@@ -1,59 +1,76 @@
+### First time setup
 
-Installing PEcAn
-----------------
+You can copy the [`env.example`](docker/env.example) file as .env in your pecan folder. The variables we want to modify are:
 
-### Virtual Machine
+* `COMPOSE_PROJECT_NAME` set this to pecan, the prefix for all containers
+* `PECAN_VERSION` set this to develop, the docker image we start with
 
-* Fastest way to get started
-* see PEcAn demo ...
+Next we will create the folders that will hold all the data for the docker containers using: `mkdir -p volumes/{lib,pecan,portainer,postgres,rabbitmq,traefik}`. The `volumes` folder will be ignored by git.
 
-### Installing from source
+First we bring up postgresql (we will start RabbitMQ as well since it takes some time to start): `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d postgres rabbitmq`. This will start postgresql and rabbitmq. We need to wait for a few minutes (you can look at the logs using `docker-compose logs postgres`) to see if it is ready.
 
-#### From GitHub
+During the time it takes for the database to start we will copy the R packages from a container to our `volumes/lib` folder. This folder contain the compiled R packages. Later we will put our newly compiled code here as well. You can copy all the data using `docker run -ti --rm -v ${PWD}/volumes/lib:/rlib pecan/base:develop cp -r /usr/local/lib/R/site-library/* /rlib/`. This will copy all compiled packages to your local machine. This only needs to be done once (or if the PEcAn base image changes drastically, for example a new version of R).
 
-```
-library(devtools)
-install_github("pecan", "PEcAnProject")
-```
+At this point postgresql should have started and is ready to load the database. This is done using `docker run -ti --rm --network pecan_pecan pecan/db`. This only needs to be done once (unless the volumes folder is removed).
 
-#### "Makefile"
+Once the database is loaded we can bring up the rest of the docker stack using `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d`. At this point you have PEcAn running in docker.
 
-```
-./scripts/build.sh -install # installs all R packages
-./scripts/build.sh -h       # list other options
+### PEcAn Development
 
-```
\ No newline at end of file
+The current folder is mounted in some containers as `/pecan`. You can see which containers exactly in `docker-compose.dev.yaml`. You can now modify the code on your local machine, or you can use [rstudio](http://localhost:8000) in the docker stack. Once you made changes to the code you can compile the code either in the terminal of rstudio (`cd pecan && make`) or using `./scripts/compile.sh` from your machine (latter is nothing more than a shell script that runs `docker-compose exec executor sh -c 'cd /pecan && make'`. 
+
+If you submit new workflows through the webinterface it will use this new code in the executor, and any other containers that have `volumes/lib` mounted inside the containers (done by `docker-compose.dev.yaml`).
```

Comment:
Hmm now does this `volumes` approach eliminate the need for rebuilding any of the images as you modify code? [Documentation](https://pecanproject.github.io/pecan-documentation/develop/docker-index.html#docker-build-images) says "rebuild pecan/base if you made changes to the PEcAn R package source code" or are these talking about different things?

Reply:
the compiled code is shared using the lib folder, so no need to build images since all containers use the same folder with packages.

Reply:
added this as well to documentation.